pkg/gofr/http/multipart_file_bind.go

```diff
@@ -146,13 +162,195 @@ func (uf *formData) setFieldValue(value reflect.Value, data string) (bool, error
 		return uf.setFloatValue(value, data)
 	case reflect.Bool:
 		return uf.setBoolValue(value, data)
-	case reflect.Invalid, reflect.Complex64, reflect.Complex128, reflect.Array, reflect.Chan, reflect.Func, reflect.Interface,
-		reflect.Map, reflect.Pointer, reflect.Slice, reflect.Struct, reflect.UnsafePointer:
-		// These types are not supported for setting via form data
+	case reflect.Slice, reflect.Array:
+		return uf.setSliceOrArrayValue(value, data)
+	case reflect.Interface:
+		return uf.setInterfaceValue(value, data)
+	case reflect.Struct:
+		return uf.setStructValue(value, data)
+	case reflect.Invalid, reflect.Complex64, reflect.Complex128, reflect.Chan, reflect.Func,
+		reflect.Map, reflect.Pointer, reflect.UnsafePointer:
 		return false, nil
+	}
+
+	return false, nil
+}
+
+func (uf *formData) setInterfaceValue(value reflect.Value, data string) (bool, error) {
+	// If the interface is not set to a concrete value, we can't modify it directly
+	if value.Kind() == reflect.Interface && value.IsNil() {
```

Comment:
the condition value.Kind() == reflect.Interface is not required as we are already checking it in the switch case.